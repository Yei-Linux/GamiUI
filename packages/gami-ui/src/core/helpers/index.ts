export * from './date-formatter'
export * from './validations.helper'
