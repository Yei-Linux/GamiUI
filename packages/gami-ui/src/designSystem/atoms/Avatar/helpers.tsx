import { TextModeType } from 'core/domain/types'
import { avatarTextModes } from './constants'
import { REGEX_RULES } from 'core/utils/constants'

export const validateMaxLetters = (
  firstLettersGot: string,
  maxSizeLetters: number
) => {
  const size = firstLettersGot.length

  if (size > maxSizeLetters) {
    return firstLettersGot.substring(0, maxSizeLetters)
  }

  return firstLettersGot
}

export const showOnlyFirstLettersOnText = (
  text: string,
  type: TextModeType,
  maxSizeLetters: number
): string => {
  const regexType = avatarTextModes[type]
  const firstLettersCaptured: string[] | null = text.match(
    REGEX_RULES[regexType]
  )

  const joinFirstLetters = firstLettersCaptured?.join('') || ''

  return validateMaxLetters(joinFirstLetters, maxSizeLetters)
}
