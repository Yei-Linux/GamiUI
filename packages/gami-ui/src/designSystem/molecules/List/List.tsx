import React from 'react'
import { IGeneralProps } from '../../../core/domain/interfaces/IGeneralProps'

import * as S from './List.styles'
import ListItem from './ListItem'

export interface IList extends IGeneralProps {
  /**
   * Content
   */
  children: React.ReactNode[]
}

const List = ({ children, ...args }: IList) => {
  return <S.List {...args}>{children}</S.List>
}

List.Item = ListItem

export default List
