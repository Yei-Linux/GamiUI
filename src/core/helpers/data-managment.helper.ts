export const mapEnum = (enumerable: any): any[] => {
  const enumMembers: any[] = Object.keys(enumerable).map(
    (key) => enumerable[key]
  )
  return enumMembers
}
