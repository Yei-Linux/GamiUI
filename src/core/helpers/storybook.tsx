import React from "react";
import TableStories from "../../styles/utilities/TableStories";
import { options as optionsStyle } from "../utils/constants";

export const getTemplate = (Component: React.ComponentType<any>): any => (
  args: any
) => {
  return <Component {...args} />;
};

export const getListTemplate = (Component: React.ComponentType<any>): any => ({
  items,
  field,
  ...args
}: {
  items: any[];
  field: any;
}) =>
  items.map((item, index) => (
    <TableStories key={index} item={item} field={field}>
      <Component {...args} {...item} />
    </TableStories>
  ));

export const stylesControl = {
  style: { control: "object" },
  textAlign: { control: { type: "select", options: optionsStyle.textAlign } },
  fontWeight: {
    control: { type: "select", options: optionsStyle.fontWeight },
  },
  width: { control: { type: "select", options: optionsStyle.width } },
  shadow: { control: { type: "select", options: optionsStyle.shadow } },
  border: { control: { type: "select", options: optionsStyle.border } },
  heigth: { control: { type: "select", options: optionsStyle.height } },
};