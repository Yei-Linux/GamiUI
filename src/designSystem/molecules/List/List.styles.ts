import styled from "@emotion/styled";
import { colorLight } from "../../../styles/theme";
import { twinStyles } from "../../../styles/utilities/twinStyles";

export const ListWrapper = styled.ul<{
  border?: string;
  shadow?: string;
  width?: string;
  heigth?: string;
}>`
  width: fit-content;
  ${(props: any) => twinStyles(props)};
`;

export const ListItemWrapper = styled.li`
  display: flex;
  justify-content: space-between;
  align-items: center;
  border-bottom: 1px solid ${colorLight.neutral.six};
  padding: 1rem;
  list-style: none;
  width: 100%;
  min-width: 250px;
`;
export const ListItemChild = styled.div`
  margin-right: 1rem;
`;