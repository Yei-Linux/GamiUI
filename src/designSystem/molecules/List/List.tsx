import React from "react";
import { IGeneralProps } from "../../../core/domain/interfaces/IGeneralProps";
import { Left, Right } from "../../../styles/utilities/flex";
import {
  SubtitleContainer,
  TitleContainer,
} from "../../../styles/utilities/text";
import {
  ListItemChild,
  ListItemWrapper,
  ListWrapper,
} from "./List.styles";

export interface IList extends IGeneralProps {
  /**
   * Content
   */
  children: React.ReactNode;
}

const List = ({ children, ...args }: IList) => {
  return <ListWrapper {...args}>{children}</ListWrapper>;
};

List.Item = ({
  children,
  avatar,
  title,
  description,
}: {
  children?: React.ReactNode;
  avatar?: React.ReactNode;
  title?: React.ReactNode;
  description?: React.ReactNode;
}) => (
  <ListItemWrapper>
    <Left>
      <ListItemChild>{avatar}</ListItemChild>
      <ListItemChild>
        <TitleContainer>{title}</TitleContainer>
        <SubtitleContainer>{description}</SubtitleContainer>
      </ListItemChild>
    </Left>
    <Right>
      <ListItemChild>{children}</ListItemChild>
    </Right>
  </ListItemWrapper>
);

List.defaultProps = {
  shadow: "SMALL",
  border: "SMALL",
};

export default List;
