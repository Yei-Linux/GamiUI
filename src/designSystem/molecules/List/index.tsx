export { default } from "./List";
export { options } from "./constants";
