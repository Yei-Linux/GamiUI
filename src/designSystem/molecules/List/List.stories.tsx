import React from "react";
import { Meta } from "@storybook/react";
import { stylesControl } from "../../../core/helpers/storybook";

import List from ".";

export default {
  title: "Molecules/List",
  component: List,
  args: { ...List.defaultProps },
  argTypes: {
    ...stylesControl,
  },
} as Meta;

export const Basic = (args: any) => (
  <List {...args}>
    <p style={{ padding: "1rem" }}>Hola como estas</p>
  </List>
);

Basic.args = {};

export const Common = (args: any) => (
  <List {...args}>
    <List.Item
      avatar={
        <img
          width="25px"
          height="25px"
          src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
        />
      }
      title="Main title"
      description="Description"
    />
  </List>
);

Common.args = {};

export const CustomContent = (args: any) => (
  <List {...args}>
    <List.Item
      avatar={
        <img
          width="25px"
          height="25px"
          src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
        />
      }
      title="Main title"
      description="Description"
    >
      <p>Ir</p>
    </List.Item>
    <List.Item
      avatar={
        <img
          width="25px"
          height="25px"
          src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
        />
      }
      title="Main title"
      description="Description"
    >
      <p>Ir</p>
    </List.Item>
  </List>
);

CustomContent.args = {};
