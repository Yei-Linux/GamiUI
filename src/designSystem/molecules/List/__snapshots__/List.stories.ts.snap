// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/List Basic 1`] = `
<ul
  className="css-1j30tb2-ListWrapper e4ps6kk0"
>
  <p
    style={
      Object {
        "padding": "1rem",
      }
    }
  >
    Hola como estas
  </p>
</ul>
`;

exports[`Storyshots Molecules/List Common 1`] = `
<ul
  className="css-1j30tb2-ListWrapper e4ps6kk0"
>
  <li
    className="css-12ytzxr-ListItemWrapper e4ps6kk1"
  >
    <div
      className="css-a72bgn-Left e1skj02x0"
    >
      <div
        className="css-iu1hhi-ListItemChild e4ps6kk2"
      >
        <img
          height="25px"
          src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
          width="25px"
        />
      </div>
      <div
        className="css-iu1hhi-ListItemChild e4ps6kk2"
      >
        <div
          className="css-y29r7q-TitleContainer etn7vk90"
        >
          Main title
        </div>
        <div
          className="css-1kp6zvx-SubtitleContainer etn7vk91"
        >
          Description
        </div>
      </div>
    </div>
    <div
      className="css-mgn85a-Right e1skj02x1"
    >
      <div
        className="css-iu1hhi-ListItemChild e4ps6kk2"
      />
    </div>
  </li>
</ul>
`;

exports[`Storyshots Molecules/List Custom Content 1`] = `
<ul
  className="css-1j30tb2-ListWrapper e4ps6kk0"
>
  <li
    className="css-12ytzxr-ListItemWrapper e4ps6kk1"
  >
    <div
      className="css-a72bgn-Left e1skj02x0"
    >
      <div
        className="css-iu1hhi-ListItemChild e4ps6kk2"
      >
        <img
          height="25px"
          src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
          width="25px"
        />
      </div>
      <div
        className="css-iu1hhi-ListItemChild e4ps6kk2"
      >
        <div
          className="css-y29r7q-TitleContainer etn7vk90"
        >
          Main title
        </div>
        <div
          className="css-1kp6zvx-SubtitleContainer etn7vk91"
        >
          Description
        </div>
      </div>
    </div>
    <div
      className="css-mgn85a-Right e1skj02x1"
    >
      <div
        className="css-iu1hhi-ListItemChild e4ps6kk2"
      >
        <p>
          Ir
        </p>
      </div>
    </div>
  </li>
  <li
    className="css-12ytzxr-ListItemWrapper e4ps6kk1"
  >
    <div
      className="css-a72bgn-Left e1skj02x0"
    >
      <div
        className="css-iu1hhi-ListItemChild e4ps6kk2"
      >
        <img
          height="25px"
          src="https://zos.alipayobjects.com/rmsportal/ODTLcjxAfvqbxHnVXCYX.png"
          width="25px"
        />
      </div>
      <div
        className="css-iu1hhi-ListItemChild e4ps6kk2"
      >
        <div
          className="css-y29r7q-TitleContainer etn7vk90"
        >
          Main title
        </div>
        <div
          className="css-1kp6zvx-SubtitleContainer etn7vk91"
        >
          Description
        </div>
      </div>
    </div>
    <div
      className="css-mgn85a-Right e1skj02x1"
    >
      <div
        className="css-iu1hhi-ListItemChild e4ps6kk2"
      >
        <p>
          Ir
        </p>
      </div>
    </div>
  </li>
</ul>
`;
